### ISSUE TITLE
error: command 'cl.exe' failed: No such file or directory

### ISSUE THREAD

**Issue by mmortazavi** [dynamsoft-dbr/python#5]
Environments: 

- Dynamsoft Barcode Reader 7.2 (license activated)
- Windows 10 
- Visual Studio 2019 (16.3.4)
- Python 3.7.3
- Opencv-python 4.1.0.25
- Numpy 1.16.4

Based on the instructions: 

- SET VS90COMNTOOLS=%VS160COMNTOOLS%
- Edited setup.py with the right paths to libs and dlls. 

Also note that I followed [this instruction](https://stackoverflow.com/questions/41724445/python-pip-on-windows-command-cl-exe-failed/41724634) to install C++ build tools for Visual Studio. Further I have set  'cl.exe' path the environment variable (`C:\Program Files (x86)\Microsoft Visual Studio\2019\BuildTools\VC\Tools\MSVC\14.23.28105\bin\Hostx64\x64`). 

Whatever I do I get the following error when running in an Anaconda prompt:
```
cd src
python setup.py build install
python3 setup.py build install
```

> running build
> running build_ext
> building 'dbr' extension
> cl.exe /c /nologo /Ox /W3 /GL /DNDEBUG /MT -IC:\Users\MajidMortazavi\Anaconda3\lib\site-packages\numpy\core\include\numpy -IC:\Users\MajidMortazavi\Anaconda3\include -IC:\Users\MajidMortazavi\Anaconda3\include /Tcdbr.c /Fobuild\temp.win-amd64-3.7\Release\dbr.obj
> error: command 'cl.exe' failed: No such file or directory

I am guessing that still I have still issues with C++ build tools for Visual Studio and Paths!!

**Comment by yushulx** [comment]
@mmortazavi  You cannot use `SET VS90COMNTOOLS=%VS160COMNTOOLS%`.  Since you've followed StackOverflow to install the `VC++ 2015.3 v140 toolset`, you should use `SET VS90COMNTOOLS=%VS140COMNTOOLS%`.

**Comment by mmortazavi** [comment]
@yushulx As mentioned in the Environments, I have `Visual Studio 2019 (16.3.4)`, that is why `SET VS90COMNTOOLS=%VS160COMNTOOLS%` was set. I did only follow ideas that were suggested in that post.

**Comment by yushulx** [comment]
@mmortazavi I know you are using VS 2019, but you cannot set the environment variable like that. It's Python's problem. Only VS2015 toolset can work.

**Comment by yushulx** [comment]
@mmortazavi I tried `SET VS90COMNTOOLS=%VS150COMNTOOLS%` for my `Visual Studio 2017`, but failed.

**Comment by mmortazavi** [comment]
@yushulx What is your suggestion for Visual Studio 2019? So for Visual Studio 2019, I have to set up the env variable like `SET VS90COMNTOOLS=%VS140COMNTOOLS%.`. 

By the way, is [pypi distribution](https://pypi.org/project/dbr/) not maintained? I can not do pip:
```
$ pip install dbr

ERROR: Could not find a version that satisfies the requirement dbr (from versions: none)
ERROR: No matching distribution found for dbr
```

**Comment by yushulx** [comment]
Yes. You can install the `Visual Studio 2015 toolset` and use `SET VS90COMNTOOLS=%VS140COMNTOOLS%`.

The maintenance of `PyPI distribution` is on hold. Please build the package with the source code. It's convenient by using the command `python setup.py build install`.

**Comment by mmortazavi** [comment]
I managed to finally install the package. Having the latest version of Visual Studio was not an issue at all. Surely I needed just to add C++ tools (located within `Desktop development with C++`) by modifying the installed Visual Studio from the installer. This is what I did from the beginning and it was not working! What I missed is the following. There are many components within `Desktop development with C++`, and I had missed checking `MSVC v140 - VS 2015 C++ build tools (v14.00)` all along. I had all other kind of built tools installed, but it turns out this is the one that has all C++ header files. This is great news. 

Yet I am running into another strange behavior. The tests work. I am able to read DataMatrix from the provided sample images at `Dynamsoft\Barcode Reader 7.2\Images`, as:

```
import os
from dbr import DynamsoftBarcodeReader

dbr = DynamsoftBarcodeReader()
key = "t0068M***"
dbr.initLicense(key)

def decodeFile(fileName):
    try:
        results = dbr.decodeFile(fileName, dbr.BF_ALL)
        if results:
            for result in results:
                print("barcode value: " + result[1])
        else:
            print('Nothing Found.')
    except Exception as err:
        print(err)

decodeFile(r'Dynamsoft\Barcode Reader 7.2\Images\DataMatrix.tif')
barcode value: DataMatrix.No.2
barcode value: DataMatrix.No.1
barcode value: DataMatrix.No.4
barcode value: DataMatrix.No.3
```

However, I am not able to read datamatrix from my own jpg images. The interesting part is that using `Dynamsoft Barcode Reader SDK` Windows Desktop version I am able to read the datamatrix for the very same images!! Is there any difference between two? I am happy to send an example of my images to see if this is a known behavior or I have issues from my end!
